### PR TITLE
Rename k8s.gcr.io to registry.k8s.io and go updates

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:

- Rename k8s.gcr.io to registry.k8s.io and go updates

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/data/infrastructure-digitalocean/cluster-template-prow-ci-version.yaml [hunk 1]
```diff
@@ -241,8 +241,8 @@ spec:
               echo "* downloading package: $$CI_URL/$$CI_CONTAINER.$$CONTAINER_EXT"
               wget "$$CI_URL/$$CI_CONTAINER.$$CONTAINER_EXT" -O "$$CI_DIR/$$CI_CONTAINER.$$CONTAINER_EXT"
               $${SUDO} ctr -n k8s.io images import "$$CI_DIR/$$CI_CONTAINER.$$CONTAINER_EXT" || echo "* ignoring expected 'ctr images import' result"
-              $${SUDO} ctr -n k8s.io images tag k8s.gcr.io/$$CI_CONTAINER-amd64:"$${CI_VERSION//+/_}" k8s.gcr.io/$$CI_CONTAINER:"$${CI_VERSION//+/_}"
-              $${SUDO} ctr -n k8s.io images tag k8s.gcr.io/$$CI_CONTAINER-amd64:"$${CI_VERSION//+/_}" gcr.io/k8s-staging-ci-images/$$CI_CONTAINER:"$${CI_VERSION//+/_}"
```
Comment (MorrisLaw) [inline test/e2e/data/infrastructure-digitalocean/cluster-template-prow-ci-version.yaml:245]: I see that in the Makefile, `us.gcr.io/k8s-artifacts-prod/cluster-api-do` was changed to `registry.k8s.io/cluster-api-do`, should the same happen for `gcr.io/k8s-staging-ci-images/$$CI_CONTAINER:"$${CI_VERSION//+/_}` or is this remaining the same?

Also, the description says `Rename k8s.gcr.io to registry.k8s.io and go updates`. The core of my question really is, 
- do all three `k8s.gcr.io`, `us.gcr.io` and `gcr.io` need to be changed
- or, just `us.gcr.io` and `k8s.gcr.io`
- or, just `k8s.gcr.io`?

Comment (cpanato) [reply test/e2e/data/infrastructure-digitalocean/cluster-template-prow-ci-version.yaml:245]: the makefile is updated as well to use registry.k8s.io when we generate the manifests for the release.


the staging registry is where we push the images when we build the main and releases and later we promote that image to the production bucket, so the `gcr.io/k8s-staging-ci-images` should stay as is, and we just need to set to use the registry.k8s.io when we generate the release manifests or when we use any image that is in the production registry besides that we are fine

Comment (MorrisLaw) [reply test/e2e/data/infrastructure-digitalocean/cluster-template-prow-ci-version.yaml:245]: Ok, perfect. Thanks for the clarification!

**Review by MorrisLaw — APPROVED** [review]
/lgtm 🚢

**Pull request merged**